src/components/common/inputs/multipleImages.tsx

```diff
@@ -0,0 +1,140 @@
+import React, { useRef, useState } from "react";
+import UploadSvg from "@/assets/svgs/UploadSvg";
+import PdfSvg from "@/assets/svgs/pdfSvg";
+import { useTranslation } from "react-i18next";
+
+export default function CommonUploadMultiplePictures({
+  label,
+  name,
+  required,
+  onChange,
+  maxSize = 10,
+}: {
+  label: string;
+  name: string;
+  required: boolean;
+  onChange: any;
+  maxSize?: number;
+}) {
+  const inputRef = useRef<HTMLInputElement>(null);
+  const [files, setFiles] = useState<any[]>([]); // To store multiple files
+  const { t } = useTranslation();
+
+  const cancelFile = () => {
+    onChange(null);
+    setFiles([]);
+  };
+
+  const handleFile = async (file: File) => {
+    const fileSizeInBytes = file.size;
+    const fileSizeInKilobytes = +(fileSizeInBytes / 1024).toFixed(1);
+    const fileSizeInMegabytes = +(fileSizeInKilobytes / 1024).toFixed(1);
+
+    if (fileSizeInMegabytes >= maxSize) {
+      alert(`File size exceeds the limit of ${maxSize} MB.`);
+      cancelFile();
+      return null;
+    }
+
+    return file;
+  };
+
+  // Remove specific file from the list
+  const removeFile = (fileToRemove: File) => {
+    const updatedFiles = files.filter((file) => file !== fileToRemove);
+    setFiles(updatedFiles);
+  };
+
+  const onChangeFile = async (e: React.ChangeEvent<HTMLInputElement>) => {
+    const selectedFiles = e.target.files ? Array.from(e.target.files) : [];
+    if (selectedFiles.length === 0) {
+      cancelFile();
+      return;
+    }
+
+    const processedFiles: any[] = [];
+    for (let file of selectedFiles) {
+      const processedFile = await handleFile(file);
+      if (!processedFile) continue;
+
+      processedFiles.push(processedFile);
+    }
+
+    // Append new files to the existing list of files and update state
+    const updatedFiles = [...files, ...processedFiles];
+    setFiles(updatedFiles);
+    onChange(updatedFiles); // Update the parent with the new list of files
+  };
+
+  const onClick = () => {
+    inputRef.current?.click();
+  };
+
+  return (
+    <div className="relative cursor-pointer" >
```

Comment:
Tons of duplicated code because we have uploadSingle image

We only need one component (remove the old one and keep this)
then a property which checks the max files allow (1, or N) or undefined for unlimeted

Reply:
Which old codes do you want me to remove? Like the reason I changed is because the remove of one image went out of the component, so now we have 2 components one where we upload the pictures and another one where we display the images.